parse.go

```diff
@@ -663,11 +663,12 @@ type scanner struct {
 }
 
 func (s *scanner) nextChar() bool {
-	if s.pos >= len(s.text) {
+	runedText := []rune(s.text)
```

Comment:
What do you think about to init runedText ones to avoid allocation and reuse it in all places where you need i rune ?

Reply:
So, i checked all usages of `scanner.text`. We can change type from `string` to `[]rune` or define new `runedText` field in `scanner` sctruct. 

The first case gives us new type and complicates upstream merging (we have to check new functions that use this field)
The last case gives us new field with `[]rune` type and it can confuse us -- "wait, why do we use single value in two different fileds with different types?". But we save compatibility with upstream.

Which one is better? (i would like to select last one)

Reply:
Do tests cover cases if we change `string` to `[]rune` ? If it doesn't break something I think we can change the type

Reply:
Tested! It's OK.

Reply:
If you want to reduce allocations further, you can use wrapper around `string` like https://pkg.go.dev/golang.org/x/exp/utf8string#String.Slice , it allows to iterate and subslice `string` as it was `[]rune`. It is O(1) on memory, but sometimes O(N) on cpu.

Reply:
Ah, i thought string type embedded in `utf8string.String`. In current case -- retyping to custom string type will make possible issues as `[]byte`